usr/src/uts/common/fs/zfs/zvol.c

```diff
@@ -1811,26 +1817,21 @@ zvol_ioctl(dev_t dev, int cmd, intptr_t arg, int flag, cred_t *cr, int *rvalp)
 
 		zfs_range_unlock(rl);
 
-		if (error == 0) {
```

Comment:
In my understanding the change you are describing would be done here, by adding && zvol_unmap_sync_enabled. What am I missing?